esmvalcore/local.py

```diff
@@ -324,13 +335,18 @@ def _select_files(filenames, timerange):
     Otherwise, the file selection occurs taking into account the time
     resolution of the file.
     """
+    timerange = str(timerange)
     if "*" in timerange:
         # TODO: support * combined with a period
         return filenames
 
     selection = []
 
     for filename in filenames:
+        start: int | str
+        end: int | str
+        start_date: int | str
+        end_date: int | str
```

Comment:
Please rename the `int` version of the variable to something else for improved readability

Reply:
Done in https://github.com/ESMValGroup/ESMValCore/pull/2772/commits/8f2f1795c1a577ee5399ceb894371eee4b3a3662